src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -71,4 +78,87 @@ object ResolverUtils {
       availableStrings: Seq[String]): Option[Seq[String]] = {
     Some(requiredStrings.map(resolve(spark, _, availableStrings).getOrElse { return None }))
   }
+
+  /**
+   * Finds all resolved strings for requiredStrings, from the given logical plan. Returns
+   * optional seq of resolved strings if all required strings are resolved, otherwise None.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param plan Logical plan to resolve against.
+   * @return Optional ListMap of tuples of resolved name string and nested state boolean
+   *         if all required strings are resolved. Else, None.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredStrings: Seq[String],
+      plan: LogicalPlan): Option[ListMap[String, Boolean]] = {
```

Comment:
Any reason to use ListMap? Seems we don't need "Map"?
How about `Seq[(String, Boolean)]`?

Reply:
Yes. `ListMap` keeps the order of the items inside it. `Seq` of tuple is harder to use. `Map.keys` gives me all the field names. Using `Seq` it would mean to use a `map` call and iterate over all items.

Reply:
I guess `keys` also iterates items & needs `toSeq` after keys. As we don't need to consider uniqueness & lookup, I think `Seq` would be better.

Reply:
@sezruby: I totally disagree, but I'll do what you requested. Bellow are my reason to disagree:

1. The code will be less readable as there will be lots of `.map(_._1)` everywhere. 
2. The `ListMap.keys` is in itself a `Set` so when you do `.keys` it will give you straight away the set collection without any iteration.